### PR TITLE
Feat/cl vault demo

### PR DESCRIPTION
**Pull request by njerschow**
## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: smart-contracts/contracts/cl-vault/src/helpers.rs [hunk 1]
```diff
@@ -187,6 +193,46 @@ pub fn with_slippage(amount: Uint128, slippage: Decimal) -> Result<Uint128, Cont
     Ok(adjusted_amount)
 }
 
+/// This function subtracts out all the reward amounts from the contract balance on chain
+pub fn get_usable_balance(
```
Comment (njerschow) [inline smart-contracts/contracts/cl-vault/src/helpers.rs:197]: @LaurensKubat to verify if this is correct

File: smart-contracts/contracts/cl-vault/src/vault/range.rs [hunk 2]
```diff
@@ -149,6 +150,14 @@ pub fn handle_withdraw_position_reply(
     let amount0 = msg.amount0;
     let amount1 = msg.amount1;
 
+    // if get_usable_balance is not something we can rely on, then this is how we will need to do it
```
Comment (njerschow) [inline smart-contracts/contracts/cl-vault/src/vault/range.rs:153]: to be removed before merge @njerschow

File: smart-contracts/contracts/cl-vault/src/vault/range.rs [hunk 3]
```diff
@@ -302,6 +316,7 @@ pub fn do_swap_deposit_merge(
         ),
     };
 
+    // TODO: slippage should be configurable by caller of modify range?
```
Comment (njerschow) [inline smart-contracts/contracts/cl-vault/src/vault/range.rs:319]: @LaurensKubat to review & provide decision

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request merged**